### DISCUSSION TITLE
Nengo DL 0.5.2 released

### DISCUSSION THREAD

**Discussion by cnrgbot** [nengo/nengo-zoo#1358] [Announcements]
> _Archived from the Nengo Discourse forum._ Originally posted by **drasmuss** on 2017-10-11. (Forum topic #401, post #1 / id 1095)

The Nengo DL team is excited to announce the release of Nengo DL 0.5.2.

### What is Nengo DL?

Nengo DL is a backend for Nengo that integrates deep learning methods (supported by the [TensorFlow](https://www.tensorflow.org/) framework) with other Nengo modelling tools. This allows users to optimize their models using deep learning training methods, improves simulation speed (on CPU or GPU), and makes it easy to insert TensorFlow models (such as a convolutional neural network) into Nengo networks.

### How do I use it?

To use Nengo DL, replace instances of `nengo.Simulator` with `nengo_dl.Simulator`.

For example, if you have a network called `model` and you run it with

```python
with nengo.Simulator(model) as sim:
    sim.run(10)
```

you would change that to

```python
with nengo_dl.Simulator(model) as sim:
    sim.run(10)
```

and that's it!

Information on accessing the more advanced features of Nengo DL can be found in the [documentation](http://www.nengo.ai/nengo_dl/).

### What's new?

0.5.2 contains a number of usability improvement that make it easier to optimize Nengo models, such as support for visualizing data in TensorBoard, variable learning rate schedules, and multi-objective training.  We've also added two new examples that demonstrate how NengoDL can be used to improve performance in more complicated SPA models (thanks to new contributor @​pblouw).  And, as always, we've fixed some bugs.  Check out the [GitHub release page](https://github.com/nengo/nengo_dl/releases) for a full changelog.

### How do I get it?

To install Nengo DL, we recommend using `pip`:

```bash
pip install nengo_dl
```

More detailed installation instructions can be found [here](https://nengo.github.io/nengo_dl/installation.html).

### Where can I learn more?

- [Documentation](http://www.nengo.ai/nengo_dl/)
- [Install from PyPI](https://pypi.python.org/pypi/nengo_dl)
- [Source on GitHub](https://github.com/nengo/nengo_dl)

### Where can I get help?

You're already there! If you have an issue upgrading or have any other questions, please post them in this forum.
